### DISCUSSION TITLE
using mouse wheel to scroll does not work as intended

### DISCUSSION THREAD

**Discussion by martijn-van-den-bergh** [clientIO/joint#2244] [Q&A] [answered]
### Introduction

I am trying to implement a way so that when the user holds the CTRL button and scrolls the mousewheel it will zoom in or out on the PaperScroller.

This seems to be going well for the first couple of ticks but after that it just goes everywhere, panning and zooming randomly it seems.

This is my code:
```
 'blank:mousewheel': (evt: joint.dia.Event, x: number, y: number, delta: number) => {
    evt.preventDefault();
    const oldscale = this.paper.scale().sx;
    const newscale = oldscale + 0.2 * delta * oldscale

    if (newscale > 0.2 && newscale < 5) {
        this.paper.scale(newscale, newscale, 0, 0);
        this.paper.translate(-x * newscale + evt.offsetX!, -y * newscale + evt.offsetY!);
    }
}

```

It also keeps scrolling though I call `preventDefault`, which i found odd.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
I suggest using the _PaperScroller_ API instead of the lower-level _Paper_ API.

```js
// This code was taken from the Database demo

paper.on('blank:mousewheel', (evt, ox, oy, delta) => {
    evt.preventDefault();
    scroller.zoom(delta * 0.2, { min: 0.4, max: 3, grid: 0.2, ox, oy });
});
```

Alternatively, you can use the `paper:pinch` and `paper:pan` events.

```js
// This code was taken from the RichTextEditor Demo

paper.on('paper:pinch', (evt, ox, oy, scale) => {
    evt.preventDefault();
    scroller.zoom(scale - 1, { min: 0.2, max: 5, ox, oy });
});

paper.on('paper:pan', (evt, tx, ty) => {
    evt.preventDefault();
    scroller.el.scrollLeft += tx;
    scroller.el.scrollTop += ty;
});
```

**Reply by martijn-van-den-bergh** [reply]
I tried your first example. It works pretty good. 

Only one problem: Sometimes during a scrollwheel tick, the browser also performs a scroll and a zoom simultaneously. Which is a bit confusing for the user.

How can I fix this? Normally the preventDefault should prevent the scroll.

**Reply by kumilingus** [reply]
The `blank:mousewheel`  is not triggered on cells. That is most likely the reason.

```js
// This code was taken from the Database demo

paper.on('blank:mousewheel', (evt, ox, oy, delta) => {
    evt.preventDefault();
    scroller.zoom(delta * 0.2, { min: 0.4, max: 3, grid: 0.2, ox, oy });
});

paper.on('cell:mousewheel', (_, evt, ox, oy, delta) => {
    evt.preventDefault();
    scroller.zoom(delta * 0.2, { min: 0.4, max: 3, grid: 0.2, ox, oy });
});
```

**Reply by martijn-van-den-bergh** [reply]
Hmm, I tried this but it's not working. It is still scrolls, but it looks like it only starts scrolling only when it reaches the max or min value of the zoom level.

Is that intended?

**Reply by kumilingus** [reply]
How to reproduce that? I've just tried the KitchenSink application, commented out the `paper:pinch` and `paper:pan` events and added the code above. All works as expected. No scrolling while using `mousewheel` detected.

**Reply by martijn-van-den-bergh** [reply]
Hmm, I will try to replicate the Kitchen sink in my app. It is based on the kitchen sink app and maybe I am doing something wrong.

**Reply by martijn-van-den-bergh** [reply, marked as answer]
If i copy this part of the kitchen sink app, it works prefectly:

```
            'paper:pinch': (_evt, ox, oy, scale) => {
                // the default is already prevented
                const zoom = this.paperScroller.zoom();
                this.paperScroller.zoom(zoom * scale, { min: 0.2, max: 5, ox, oy, absolute: true });
            }

```

Thanks for the tip!
